activerecord/lib/active_record/associations/belongs_to_association.rb

```diff
@@ -77,6 +77,8 @@ def replace(record)
             raise_on_type_mismatch!(record)
             set_inverse_instance(record)
             @updated = true
+          elsif inverse_name = reflection.inverse_of&.name
+            target&.association(inverse_name)&.reset
```

Comment:
I think we can reuse some existing methods for looking up the inverse.
Maybe the following?
```suggestion
          elsif inverse = inverse_association_for(target)
            inverse.reset
```

Or maybe even extract a method and make it symmetrical to `set_inverse_instance`
```suggestion
else
  reset_inverse_for(target)
```